### PR TITLE
linting of lp2p

### PR DESCRIPTION
**Pull request by willscott**
Note: linting complains that our protocol buffer go code was being generated by a deprecated version of the protocol buffer tool / api.

This PR upgrades protocol buffers. if the generated fields weren't manually modified, this should we wire compatible, but there's a potential that in regeneration / upgrade an incompatibility was introduced.

The good news is that one can now type `go generate` from the `protobuf/` directory to re-generate the protocol buffer generated code from the .proto definitions. Next, we'll add a CI check to confirm that generated files always remain in-sync with the source definitions :)

### CODE REVIEW

File: protobuf/crypto/share.proto [hunk 1]
```diff
@@ -4,8 +4,8 @@
  */
 syntax = "proto3";
 
-/*option go_package = "github.com/drand/drand/protobuf/crypto";*/
-option go_package = "share";
+option go_package = "github.com/drand/drand/protobuf/crypto";
```
Comment (alanshaw) [inline protobuf/crypto/share.proto:7]: should be protobuf/share?

Comment (willscott) [reply protobuf/crypto/share.proto:7]: I supposed what this means is nothing uses this file.

the directory is 'protobuf/crypto', and that package is what will cause the generated .go file to end up in the right place

Seems like the right answer is to remove the proto / generated file here entirely though?

Comment (alanshaw) [reply protobuf/crypto/share.proto:7]: I don’t know enough about this yet to make that call 😢 but I agree with your assertion

**Comment by willscott** [comment]
removed entirely and everything still seems happy. looking for someone to approve

**Pull request merged**